engine/src/main/java/org/terasology/physics/engine/PhysicsSystem.java

```diff
@@ -242,13 +244,55 @@ public void update(float delta) {
 
         for (CollisionPair pair : collisionPairs) {
             if (pair.b.exists()) {
-                pair.a.send(new CollideEvent(pair.b,pair.pointA,pair.pointB,pair.distance,pair.normal));
+                short bCollisionGroup = getCollisionGroupFlag(pair.b);
+                short aCollidesWith = getCollidesWithGroupFlag(pair.a);
+                if((bCollisionGroup & aCollidesWith) != 0 || 
+                        (pair.b.hasComponent(BlockComponent.class) && !pair.a.hasComponent(BlockComponent.class))){
```

Comment:
No, i check for blocks here in addition to the entities/items (in line 249). I just added this here for blocks because i don't know if they have collision filter or not and this bypasses filters for blocks and sends CollideEvent only to the other entity that is colliding with block and no event is sent to block entities.